### ISSUE TITLE
Publish OSX Build // Create OSX Build Instructions

### ISSUE THREAD

**Issue by Miserlou** [TASVideos/BizHawk#1017]
It'd be great if there were a formal OSX build of this project. I found one floating around the internet that seems to "work" when Mono is installed, but it can't actually run any games as far as I can tell, and I have no idea when it was built, and for what platform. I have some ideas that I'd like to contribute to this codebase, but I'd like to get the OSX build situation formalized before I work on any of that.

Would anybody be interested in working with me on creating formal OSX build instructions, with the intention of producing an OSX-specific build for general publication?

Thanks, and great work!

**Comment by Sappharad** [comment]
http://tasvideos.org/forum/viewtopic.php?t=12659

You're welcome to port 2.0 if you want to.

**Comment by Miserlou** [comment]
Nice, thanks for that resource! Shame about the need to port to 2.0, which looks like a real monster undertaking.

As an alternative, we could decide on a final 1.x release and then fork and simply keep adding cores (N64, PSX, etc) to that branch rather than try to keep up to what's going on with the 2.0 branch. It's not a very pretty solution but I think it would create the most amount of overall utility.

**Comment by Sappharad** [comment]
The last 1.x release was somewhat reasonable, I don't see a need to do another one. All of the cores were supported except for N64 and Saturn. There wouldn't be a need to fork anything, the mono-portable branch has not been deleted since it hasn't been replaced yet.

Adding Saturn now would be pointless though, because 2.0 has a completely different Saturn core that is much better and the old one was pretty bad so nobody should be using it anyway. N64 just had a lot of parts with its whole plugin system, and I don't remember if there were other problems preventing it from happening. SNES wasn't supported for a long time because of named pipes, but eventually they re-wrote that so I was able to get it working in one of the later 1.x releases.

As for 2.0, I'm stuck in the same place I was a month ago and I'm still trying to troubleshoot that for a few hours a week. It needs to go 64-bit since the Windows builds did, and especially since Apple's dropping 32-bit support in 2019. But WinForms doesn't support 64-bit, so a new UI has to be written instead of re-using the Windows one. That's fine, but I also need to build against Xamarin.Mac instead of MonoMac to target 64-bit APIs and OpenTK has a compatibility issue with that.
https://github.com/opentk/opentk/issues/647

I've basically reached the point where I'm constantly hitting compatibility limitations that weren't a problem before and I don't enjoy trying to fix them. I think the OpenTK problem is a Microsoft bug, the linker either completely ignores OpenTK or it tries to link both OpenGL and OpenGL ES and fails because desktops don't have the mobile API, but I haven't gotten far enough in troubleshooting the problem to confirm that yet.

In short, 1.x is pretty good, despite being a bit awkward and limited at times. But I don't see a reason to go forward on that given 2.0. But 2.0 is stuck too, for the reasons previously stated. Things worked better when I could just bring over the latest changes from the Windows version in a few hours and slowly add features that were missing every few months. If I abandoned some other side projects I'd have more time for this one, but I don't want to do that.

**Comment by Sappharad** [comment]
I'm actually going to close this, unless someone else minds and wants it re-opened. The two things you brought up were basically addressed by linking to that forum thread.

1. I posted all of the macOS builds in that thread. They can't be posted here on GitHub because of limitations to how its release system works. It basically allows 1 binary per release, which is tied to a specific revision of master. (As far as I know) Keeping the macOS builds partially hidden by only posting them in that thread is basically my fault, if you've used them you probably know there's a handful of features that just don't work like TASStudio and Lua, and I didn't bother blocking off the non-working stuff. It's generally pretty good as an emulator to play games on, but not much more than that. I would probably start doing a better job providing awareness of it once the UI is replaced so there aren't untested options laying around.

2. Build instructions were also mentioned somewhere in that thread at some point, but here's the basic run-down:
- Install Visual Studio for Mac
- Download a copy of the mono-portable branch:
https://github.com/TASVideos/BizHawk/tree/mono-portable
At this point though, the portable branch has been untouched since 2.0 came out.
- Open either BizHawk.sln or MonoMacWrapper.sln in Visual Studio. (It's your choice. The Mac wrapper is what actually gets released, but if you want the WinForms UI completely unmodified then BizHawk.sln will work)
- Build the code and run it.

The only error you'll probably run into is with the Version project complaining about svnrev.cs missing. It's supposed to run a shell script to generate Version/svnrev.cs but for some reason the script never actually gets executed no matter how I'd put it in there. So as a work-around if it fails just copy the template in the folder and make your own copy of that file. Outside of the missing file, it's supposed to be that easy. You just build the code and it works.

**Comment by Miserlou** [comment]
Excellent, thank you for taking the time to document this.

**Comment by vadosnaprimer** [comment]
@Sappharad you can attach mac build packages to usual bizhawk releases.
